azure-pipelines.yml

```diff
@@ -22,18 +22,24 @@ steps:
     mkdir -p '$(modulePath)'
     shopt -s extglob
     shopt -s dotglob
-    mv !(gopath) '$(modulePath)'
+    mv !(work) '$(modulePath)'
     echo '##vso[task.prependpath]$(GOBIN)'
     echo '##vso[task.prependpath]$(GOROOT)/bin'
   displayName: 'Set up the Go workspace'
-
 - script: |
+    set -e
     go version
-    go get -v -t -d ./...
-    if [ -f Gopkg.toml ]; then
-        curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
-        dep ensure
-    fi
-    go build -v .
+    curl -sSL https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+    dep ensure -v
+    go get -u golang.org/x/lint/golint
+  workingDirectory: '$(modulePath)'
+  displayName: 'Install Dependencies'
+- script: go vet $(go list ./... | grep -v vendor)
+  workingDirectory: '$(modulePath)'
+  displayName: 'Vet'
+- script: go build -v $(go list ./... | grep -v vendor)
+  workingDirectory: '$(modulePath)'
+  displayName: 'Build'
+- script: make test
```

Comment:
I suggest we remove this step temporarily, since this script `test/test.sh` needs some modification to fit the ADO